app/models/mets_document.rb

```diff
@@ -9,8 +9,23 @@ def initialize(mets_file)
     @mets = File.open(@source_file) { |f| Nokogiri::XML(f) }
   end
 
+  # Takes a string of xml
+  # def initialize(mets_string)
+  #   @source_string = mets_string
+  #   @mets = Nokogiri::XML(@source_string)
+  # end
+
   def oid
-    @mets.xpath("//goobi:metadata[@name='CatalogIDDigital']").first.content.to_s
+    catalog_id_digital = @mets.xpath("//goobi:metadata[@name='CatalogIDDigital']").first
+    return unless catalog_id_digital
+    catalog_id_digital.content.to_s
```

Comment:
would
`catalog_id_digital = @mets.xpath("//goobi:metadata[@name='CatalogIDDigital']").first&.content&.to_s`
being you back to a one liner w/o changing clarity?